### PR TITLE
fix selectbox bug

### PR DESCRIPTION
**Pull request by avrilpearl**
Fixes bug relating to selectbox atom in mayflower react

### CODE REVIEW

File: react/src/components/atoms/forms/SelectBox/index.js [hunk 1]
```diff
@@ -46,7 +46,7 @@ class SelectBox extends React.Component {
     const labelClassNames = stackLabel ? 'ma__select-box__label' : 'ma__label--inline ma__label--small';
     const selectBoxInline = stackLabel ? '' : 'ma__select-box__field--inline';
     const selectedIndex = this.props.options.findIndex((option) => option.text === selected);
```
Comment (clairesunstudio) [inline react/src/components/atoms/forms/SelectBox/index.js]: this still needs work, if the value selected from props is not in this.props.options, selectedIndex is undefined and component will break

File: react/src/components/atoms/forms/SelectBox/index.js [hunk 2]
```diff
@@ -45,8 +45,11 @@ class SelectBox extends React.Component {
     const { stackLabel } = this.props;
     const labelClassNames = stackLabel ? 'ma__select-box__label' : 'ma__label--inline ma__label--small';
     const selectBoxInline = stackLabel ? '' : 'ma__select-box__field--inline';
-    const selectedIndex = this.props.options.findIndex((option) => option.text === selected);
-
+    const selectedValue = this.props.options.map((option) => {
+      if (option.text === selected) {
+        return option.value;
+      } return this.props.options[0].value;
+    });
```
Comment (pmallett-mc) [inline react/src/components/atoms/forms/SelectBox/index.js]: This assigns an array to `selectedValue` (one with as many elements as this.props.options has). Probably not what was intended.

**Review by pmallett-mc — DISMISSED** [review]
This isn't quite right -- see the example here https://reactjs.org/docs/forms.html#the-select-tag; the component state should be tracking the select's value, not the text that is displayed. Then we don't have to look up into the options list to find ids or keys, only check whether the value exists or not for validation.

**Comment by avrilpearl** [comment]
@pmallett-mc  --> for the selectbox, the text was the state we originally started tracking and are dependent on in eec and massgov-search. I get that at some point we should switch this over to value but that is not what this pr is trying to resolve so that can be a follow up pr that will require work in massgov-search and eec. this pr is just trying to fix a bug that is blocking eec work.

**Comment by clairesunstudio** [comment]
this will fix situations when there's no match, the text rendered and value are kept in sync:
e.g.

screen shot 2018-09-28 at 3 01 18 pm
screen shot 2018-09-28 at 3 00 00 pm

**Review by clairesunstudio — APPROVED** [review]

**Pull request merged**